modmail/config.py

```diff
@@ -1,144 +1,740 @@
-import asyncio
-import datetime
-import json
+import functools
+import inspect
 import logging
 import os
-import sys
+import pathlib
+import types
 import typing
-from pathlib import Path
-from typing import Any, Dict, Optional, Tuple
+from collections import defaultdict
 
+import attr
+import desert
 import discord
-import toml
-from discord.ext.commands import BadArgument
-from pydantic import BaseModel
-from pydantic import BaseSettings as PydanticBaseSettings
-from pydantic import Field, SecretStr
-from pydantic.color import Color as ColorBase
-from pydantic.env_settings import SettingsSourceCallable
-from pydantic.types import conint
-
-
-log = logging.getLogger(__name__)
-
-CONFIG_PATHS: list = [
-    f"{os.getcwd()}/config.toml",
-    f"{os.getcwd()}/modmail/config.toml",
-    "./config.toml",
+import discord.ext.commands
+import discord.ext.commands.converter
+import marshmallow
+import marshmallow.fields
+import marshmallow.validate
+
+from modmail.log import ModmailLogger
+
+
+_log_optional_deps = logging.getLogger("modmail.optional_dependencies")
+_silence = " You can silence these alerts by ignoring the `modmail.optional_dependencies` logger"
+try:
+    import dotenv
+except ModuleNotFoundError:  # pragma: nocover
+    _log_optional_deps.notice("dotenv was unable to be imported." + _silence)
```

Comment:
Not worth it since silence is always at the end.

[Per this comment](https://github.com/discord-modmail/modmail/pull/75#discussion_r769234612), I may be removing that warning anyways.